src/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs

```diff
@@ -170,6 +164,24 @@ private async Task<CreationResultStatus> CreateTemplateAsync(ITemplateMatchInfo
                 return CreationResultStatus.CreateFailed;
             }
 
+            string fallbackName = new DirectoryInfo(_commandInput.OutputPath ?? Directory.GetCurrentDirectory()).Name;
+
+            if (string.IsNullOrEmpty(fallbackName) || string.Equals(fallbackName, "/", StringComparison.Ordinal))
+            {   // DirectoryInfo("/").Name on *nix returns "/", as opposed to null or "".
+                fallbackName = null;
+            }
+
+            // Name returns <disk letter>:\ for root disk folder on Windows - replace invalid chars
+            if (fallbackName.IndexOfAny(invalidChars) > -1)
```

Comment:
if the statement in line 171 is reached, this line would throw.  Is this by design?